x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_readonly/fields/severity_mapping/severity_mapping.stories.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { Story } from '@storybook/react';
+import { FieldReadOnly } from '../../field_readonly';
+import type { DiffableAllFields } from '../../../../../../../../../common/api/detection_engine';
+import { SeverityMappingReadOnly } from './severity_mapping';
+
+export default {
+  component: SeverityMappingReadOnly,
+  title:
+    'Rule Management/Prebuilt Rules/Upgrade Flyout/ThreeWayDiff/FieldReadOnly/severity_mapping',
+};
+
+interface TemplateProps {
+  finalDiffableRule: Partial<DiffableAllFields>;
+}
+
+const Template: Story<TemplateProps> = (args) => {
+  return (
+    <FieldReadOnly
+      fieldName="severity_mapping"
+      finalDiffableRule={args.finalDiffableRule as DiffableAllFields}
+    />
+  );
+};
+
+export const Default = Template.bind({});
+
+Default.args = {
+  finalDiffableRule: {
+    severity_mapping: [
+      {
+        field: 'event.severity',
+        operator: 'equals',
+        severity: 'low',
+        value: 'not severe',
+      },
+      {
+        field: 'event.severity',
+        operator: 'equals',
+        severity: 'high',
+        value: 'pretty severe',
```

Comment:
<img width="396" alt="Screenshot 2024-09-05 at 2 37 03 PM" src="https://github.com/user-attachments/assets/9950fd40-ae60-4a1b-b19a-5062c9702be9">

The `value` field here is I think always (?) a number, the component itself doesn't seem to handle fields longer than that. If we do have to handle strings it might be worthwhile updating the `max-width` in the `SeverityMappingItem` columns but other than that might be a more accurate representation to have these as numbers

Reply:
I've checked our prebuilt rules and found that in some cases these can be strings. This [rule](https://github.com/elastic/integrations/blob/3afb3ba366190837443219a47c38ff3d2c3da051/packages/security_detection_engine/kibana/security_rule/f1a6d0f4-95b8-11ed-9517-f661ea17fbcc_1.json#L48), for example.

I made the value column a bit wider and fixed the broken dot-dot-dot ellipsis in this [commit](https://github.com/nikitaindik/kibana/commit/e50ece0b077172eee47edec6eb5e51959da21c93).

https://github.com/user-attachments/assets/13d60668-f270-4430-80f4-d0c091e2e55a